src/lib/homestore_backend/hs_shard_manager.cpp

```diff
@@ -183,15 +199,33 @@ void HSHomeObject::do_shard_message_commit(int64_t lsn, ReplicationMessageHeader
     auto shard_info = deserialize_shard_info(r_cast< const char* >(value.bytes), value.size);
     switch (header.msg_type) {
     case ReplicationMessageType::CREATE_SHARD_MSG: {
-        add_new_shard_to_map(std::make_unique< HS_Shard >(shard_info, blkids.chunk_num()));
+        bool shard_exist = false;
+        {
+            std::scoped_lock lock_guard(_shard_lock);
+            shard_exist = (_shard_map.find(shard_info.id) != _shard_map.end());
+        }
+
+        if (!shard_exist) { add_new_shard_to_map(std::make_unique< HS_Shard >(shard_info, blkids.chunk_num())); }
+
         break;
     }
 
     case ReplicationMessageType::SEAL_SHARD_MSG: {
-        auto chunk_id = get_shard_chunk(shard_info.id);
-        RELEASE_ASSERT(chunk_id.has_value(), "Chunk id not found");
-        chunk_selector()->release_chunk(chunk_id.value());
-        update_shard_in_map(shard_info);
+        ShardInfo::State state;
+        {
+            std::scoped_lock lock_guard(_shard_lock);
+            auto iter = _shard_map.find(shard_info.id);
+            RELEASE_ASSERT(iter != _shard_map.end(), "Missing shard info");
+            state = (*iter->second)->info.state;
+        }
+
+        if (state == ShardInfo::State::OPEN) {
+            auto chunk_id = get_shard_chunk(shard_info.id);
+            RELEASE_ASSERT(chunk_id.has_value(), "Chunk id not found");
+            chunk_selector()->release_chunk(chunk_id.value());
```

Comment:
here is also the  similar problem as mentioned above.

no need to call `release_chunk` , just do not put it in to `excludingChunks` and then `build_per_dev_chunk_heap `